### DISCUSSION TITLE
Use Grammar to parse from application

### DISCUSSION THREAD

**Discussion by evangeloszotos** [eclipse-langium/langium#1136] [General]
Is it possible to define my DSL with langium, then to build the parser and use the parser from inside of my own application?

For example I have a nodejs application and a user can input a string.
I  would like to parse that string and get the AST for further processing. 

Without the need of using an Editor such like VSCode.

```ts
import langium from "langium"

const myParser = langium.createFromGrammar(myDslGrammar)

const ast = myParser.parse("some valid string")
```

Some way, to use the created parser programmatically to parse a string, wether on the server or on the client.

**Comment by msujew** [comment]
Hey @evangeloszotos,

that can be easily done, see our [cli examples](https://github.com/langium/langium/blob/main/examples/domainmodel/src/cli/cli-util.ts#L14-L40) or the [lox example](https://github.com/langium/langium-lox/blob/main/src/interpreter/runner.ts). The easiest way to get started with something is by running the Langium yeoman generator and generating a project that only contains the language and an associated CLI. The code used for the CLI is just any normal JS code that can be embedded in frontend or backend code without any extras.

You can probably skip the stuff related to the `DocumentBuilder` if you only want to parse stuff and use the `LangiumParser` directly. However, if your language contains cross references between elements, those only resolve if you run the full document lifecycle via the `DocumentBuilder`. See [here](https://langium.org/docs/document-lifecycle/) for more information on that topic.

**Reply by evangeloszotos** [reply]
Thank you

I ll check it out
